pyro/distributions/spanning_tree.py

```diff
@@ -112,20 +112,21 @@ def log_partition_function(self):
         # use a Cholesky decomposition to compute the log determinant.
         # See https://en.wikipedia.org/wiki/Kirchhoff%27s_theorem
         V = self.num_vertices
-        grid = make_complete_graph(V)
-        shift = self.edge_logits.detach().max()
-        edge_probs = (self.edge_logits - shift).exp()
```

Comment:
these are normalized along each column/row?

Reply:
That's right, I'm normalizing by column, but since the matrix is symmetric, I split that normalization to be half along rows and half along columns. The diagonal doesn't care whether it's all-column or all-row or half-and-half.